executor/common.h

```diff
@@ -207,8 +208,19 @@ static void use_temporary_dir(void)
 static void remove_dir(const char* dir)
 {
 	DIR* dp = opendir(dir);
-	if (dp == NULL)
+	if (dp == NULL) {
+		if (errno == EACCES) {
+			// We could end up here in a recursive call to remove_dir() below.
+			// One of executed syscall could end up creating a directory rooted
+			// in the current working directory created by loop() with zero
+			// permissions. Try to perform a best effort removal of the
+			// directory.
+			if (rmdir(dir))
+				exitf("rmdir(%s) failed", dir);
+			return;
+		}
 		exitf("opendir(%s) failed", dir);
```

Comment:
How valuable is this diagnostics? I'm thinking of making this code more assertive:
```C
DIR* dp = opendir(dir);
struct dirent* ep = 0;
while (dp != NULL && (ep = readdir(dp))) {
   // remove the contents if you can
}
// Cleanup if open succeeded.
if (dp != NULL)
	closedir(dp);
// Try to delete no matter what.
if (rmdir(dir))
               // Dunno what the caller expectations are, maybe they are OK with best effort
               // removal and we should complain when any of the unlinks failed?
		exitf("rmdir(%s) failed", dir);
```